v2/tools/asoctl/internal/importing/not_importable_error.go

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"fmt"
+)
+
+// NotImportableError is an error that indicates that a resource cannot be imported for a reason we know about.
+// This allows us to continue the import even if some expected errors occur.
+type NotImportableError struct {
+	Name    string
+	Because string
+}
+
+// Ensure we implement the error interface
+var _ error = NotImportableError{}
+
+func NewNotImportableError(name string, because string) NotImportableError {
```

Comment:
super minor: Our standard style is that `New` returns a ptr, although in this case it really doesn't seem like it matters much. You could make this type impl error as a pointer and return a pointer, it technically would save some string copies on the return (I think?)

I did a quick check and as far as I can tell most of our errors do use a ptr type (see `NotOwnedError`), though I am not sure if there's any great reason for that or if it's just because I wrote most of them, lol.

Reply:
Changed to a pointer.